h/views/account_signup.py

```diff
@@ -82,10 +78,13 @@ def validation_failure(self):
     def js_config(self) -> dict[str, Any]:
         return {"csrfToken": get_csrf_token(self.request)}
 
-    def redirect_if_logged_in(self):
-        if self.request.authenticated_userid is not None:
-            raise httpexceptions.HTTPFound(
-                self.request.route_url(
-                    "activity.user_search", username=self.request.user.username
-                )
-            )
+
+# It's possible to try to sign up while already logged in. For example: start
+# to signup but don't submit the final form, then open a new tab and log in,
+# then return to the first tab and submit the signup form. This view is called
+# in these cases.
+@view_config(route_name="signup", is_authenticated=True)
+def is_authenticated(request):
+    return HTTPFound(
+        request.route_url("activity.user_search", username=request.user.username)
+    )
```

Comment:
This view function _could_ have been a method in the `SignupViews` class above and then it wouldn't need the `route_name="signup"` as that'd be inherited from the class's `@view_defaults`. But in future new routes, handled by new view classes, are also going to make use of this redirect-if-already-authenticated view so it's going to make sense for it to be a standalone function.